### DISCUSSION TITLE
Temperatur-Anzeige besser visualisieren

### DISCUSSION THREAD

**Discussion by nk-henning** [solectrus/solectrus#2908] [Ideas]
Ahoi!

Ich muss die Betriebstemperatur meines Speichers etwas im Auge behalten. Die derzeitige Temperaturanzeige "warnt" mich visuell leider nicht, wenn die Temp zu tief oder zu hoch ist. 

Man prüfe bitte die Notwendigkeit, die Anzeige visuell zu optimieren. Meine Idee wäre: Erreichen der unteren Grenz-Temperatur: blau, optimaler Bereich: grau (oder grün), Erreichen der kritischen Temperatur: rot. Das Ganze in schattierten Abstufungen.

Vielleicht in etwa so.
<img width="430" alt="Bildschirmfoto 2024-03-11 um 22 04 24" src="https://github.com/solectrus/solectrus/assets/122367820/77f94479-5b4a-4586-a5ae-0698b15ce90a">

Herzliche Grüße
Henning

**Comment by ledermann** [comment]
Ja, das ist definitiv sinnvoll. Bei der Autarkie gibt es bereits eine wert-abhängige Einfärbung, für die Gehäusetemperatur ist das ebenfalls sinnvoll. 

Zwangssläufig erfordert dies eine individelle Festlegung der Grenzwerte (Min und Max). In den Datenblättern der Speicher finden sich zwar die zulässigen und optimalen Umgebungstemperaturen, die Gehäusetemperatur ist aber wieder etwas anderes. Außerdem unterstützt SOLECTRUS mittlerweile nicht nur SENEC, sodass hier sowieso nicht mit konstanten Grenzen gearbeitet werden kann.

#2915 ist angelegt.

Es gibt übrigens auch ein Issue für eine Alarmierung bei Überschreiten der Höchsttemperatur (#1851), aber das ist noch ein ganz anderes Thema und unabhängig von der Visualisierung.

**Reply by StevieC121176** [reply]
Evtl. wäre diese wertabhängige Einfärbung auch beim SOC der Batterie möglich. z.Bsp. unter 10% = rot

**Reply by ledermann** [reply]
Ja, denkbar. Es sollte aber ein Weg gefunden werden, der die Wichtigkeit der jeweiligen Zustände berücksichtigt. Ein leerer Akku im Winter oder frühmorgens ist völlig normal und erfordert keine Aktion, bei einer zu hohen Gehäusetemperatur aber muss etwas getan werden. Es sollte daher nicht beides auf die gleiche Weise visualisiert werden.

Dass derzeitig allein die Autarkie eingefärbt wird, ist übrigens auch nicht so ideal.

Vielleicht so:
https://play.tailwindcss.com/TUSodJzNdE

**Reply by StevieC121176** [reply]
Wow, Ja das gefällt mir auch sehr gut. Ich finde das mit den Zwischenfarben gut, quasi zu hoch, Wohlfühlbereich und zu niedrig.Auch die Animationen, top.
Da bin ich gespannt drauf

**Comment by nk-henning** [comment]
Das Demo sieht sehr gut aus. Auf die Animation kann man m.E. Verzichten.

Der Einfachheit halber sollte man auch auf die zwischenfärbungen
verzichten. Grenzwert zu blau und Grenzwert zu rot einstellen und fertig.
Damit würde der Augenmerk jedenfalls deutlich besser realisiert werden.

**Reply by ledermann** [reply]
Ja, ich denke auch, dass die Zwischenfärbungen nicht erforderlich sind.

Bei Animationen denke ich, dass das ganze Dashboard natürlich nicht ständig herumzappeln darf. Eine Temperatur außerhalb der Norm ist aber eine (seltene) Ausnahmesituation, bei der man aktiv werden muss. Da finde ich das angemessen. Eine Möglichkeit wäre, Animationen grundsätzlich per [prefers-reduced-motion](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion) zu deaktivieren.
